src/mlpack/methods/ann/layer/one_hot_layer.hpp

```diff
@@ -58,7 +58,7 @@ class OneHotLayer
     output = inputActivations;
     output.zeros();
 
-    arma::uword maxIndex;
+    arma::uword maxIndex = 0;
```

Comment:
Yeah; I agree that the -Wuninitialized is erroneous in this case, but I am not sure there is any other way to stop that warning from appearing here other than initializing the index to 0.  Do you know any other ways?